File: leetcode/findSmallestSetOfVertices.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @return {number[]}
 */
var findSmallestSetOfVertices = function (n, edges) {
  const noInVertexes = new Set();
  for (let i = 0; i < n; i++) {
    noInVertexes.add(i);
  }
  for (const [u, v] of edges) {
    noInVertexes.delete(v);
  }
  return Array.of(...noInVertexes);
};
findSmallestSetOfVertices(3, [
  [1, 2],
  [1, 0],
  [0, 2],
]);

File: leetcode/findSmallestSetOfVertices.py
class Solution:
    def findSmallestSetOfVertices(
            self, n: int, edges: List[List[int]]) -> List[int]:
        # zero_in_degree_vertex = set(range(n))
        # for u, v in edges:
        #     zero_in_degree_vertex.discard(v)
        # return zero_in_degree_vertex
        return set(range(n)) - set(v for u, v in edges)
